ViewModels/BluetoothViewModel.cs

```diff
@@ -290,6 +399,108 @@ private async Task<bool> doScanAsync()
+        {
+            Context con = Android.App.Application.Context;
+            UsbManager manager = (UsbManager)con.GetSystemService(Context.UsbService);
+
+            var features = con.PackageManager.GetSystemAvailableFeatures();
+            foreach ( var feature in features ) 
+            {
+                logger.info("{0} feature available", feature.Name);
+            }
+
+            if (manager.DeviceList.Count == 0)
+            {
+                logger.info("No USB devices found");
+            }
+
+            foreach (Android.Hardware.Usb.UsbDevice acc in manager.DeviceList.Values)
+            {
+                this.acc = acc;
+
+                String desc = String.Format("Vid:0x{0:x4} Pid:0x{1:x4} (rev:{2}) - {3}",
+                    acc.VendorId,
+                    acc.ProductId,
+                    acc.Version,
+                    acc.DeviceName);
+
+                logger.info("found usb device {0}", desc);
+
+                if (acc.VendorId == 0x24aa)
+                {
+                    USBViewDevice uvd = new USBViewDevice(acc.DeviceName, acc.VendorId.ToString("x4"), acc.ProductId.ToString("x4"));
+                    usbDeviceList.Add(uvd);
+
+                    usbIntent = PendingIntent.GetBroadcast(con, 0, new Intent(ACTION_USB_PERMISSION), PendingIntentFlags.Immutable);
+                     
+                    //LibUsbDotNet.UsbDevice usbDevice = LibUsbDotNet.UsbDevice.OpenUsbDevice(d => d.Pid == acc.ProductId);
+                    manager.RequestPermission(acc, usbIntent);
+                    
+                }
+
+            }
+        }
+        catch (Exception ex)
+        {
+            logger.info("USB grab failed with error {0}", ex.Message);
+        }
+
+        /*
+        try
+        {
+            UsbRegDeviceList deviceRegistries = UsbDevice.AllDevices;
+            if (deviceRegistries == null)
+            {
+                logger.info("No USB devices found");
+            }
+            else if (deviceRegistries.Count == 0)
+            {
+                logger.info("No USB devices found");
+            }
+
+            else
+            {
+                foreach (UsbRegistry usbRegistry in deviceRegistries)
+                {
+                    String desc = String.Format("Vid:0x{0:x4} Pid:0x{1:x4} (rev:{2}) - {3}",
+                        usbRegistry.Vid,
+                        usbRegistry.Pid,
+                        (ushort)usbRegistry.Rev,
+                        usbRegistry[SPDRP.DeviceDesc]);
+
+                    logger.info("attempting to open {0}", desc);
+
+                    USBViewDevice uvd = new USBViewDevice("Test", usbRegistry.Vid.ToString("x4"), usbRegistry.Pid.ToString("x4"));
+                    usbDeviceList.Add(uvd);
+                }
+            }
+        }
+        catch (Exception ex)
+        {
+            logger.info("USB grab failed with error {0}", ex.Message);
+        }
+        */
+
+        return true;
+        
+    }
```

Comment:
This function packs a lot of punch. Nice!

Reply:
It's where the magic happens for sure